### PR TITLE
TM-873 Command for creating mocks from database fixtures.

### PR DESCRIPTION
**Pull request by lyzkov**

### CODE REVIEW

File: src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     * @var GeneratorInterface
+     */
+    private $generator;
+
+    /**
+     * @var ContainerRepository
+     */
+    private $containerRepository;
+
+    public function __construct(Filesystem $filesystem, GeneratorInterface $generator, ContainerRepository $containerRepository)
```
Comment (kallosz) [inline src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php]: missing docs.

File: src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+     * @var TagTreeBuilderInterface
+     */
+    private $treeBuilder;
+
+    /**
+     * @var ContainerRepository
+     */
+    private $containerRepository;
+
+    public function __construct(Filesystem $filesystem, TagTreeBuilderInterface $treeBuilder, ContainerRepository $containerRepository)
```
Comment (kallosz) [inline src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php]: missing docs.

File: src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            ->setName('container:generate:tag-tree')
+            ->addArgument('access-id', InputArgument::REQUIRED, 'Access id of container that should be mocked.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $accessId = $input->getArgument('access-id');
+        $container = $this->containerRepository->findLastPublishedVersionByAccessId($accessId);
+
+        $file = <<<EOF
```
Comment (kallosz) [inline src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php]: this should be moved to template to increase readability in this file.

**Comment by michalsikora** [comment]
missing tests for command classes

**Comment by kallosz** [comment]
@michalsikora  +1.

**Comment by bua89** [comment]
@michalsikora +1.

File: app/config/parameters.yml.dist [hunk 4]
```diff
@@ -28,6 +28,7 @@ parameters:
     seventag_domain: 'http://7tag.dev'
     seventag_javascript_library_path: '%kernel.root_dir%/Resources/container-lib/library.js.dist'
     seventag_javascript_cache_dir: '%kernel.root_dir%/../web/containers'
+    seventag_javascript_tests_dir: '%kernel.root_dir%/../tests'
```
Comment (kallosz) [inline app/config/parameters.yml.dist]: this shouldn't be in parameters.yml 
We don't need to change this path depends on different server or instance.

File: src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+     * @var ContainerRepository
+     */
+    private $containerRepository;
+
+    /**
+     * @param Filesystem $filesystem
+     * @param GeneratorInterface $generator
+     * @param ContainerRepository $containerRepository
+     */
+    public function __construct(Filesystem $filesystem, GeneratorInterface $generator, ContainerRepository $containerRepository)
```
Comment (michalsikora) [inline src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php]: to many dependents here

this logic should be in separete service
```php
        $accessId = $input->getArgument('access-id');
        $container = $this->containerRepository->findByAccessId($accessId);

        $this->filesystem->write(
            sprintf('container-debugger/tests/mock/stg.js', $container->getAccessId()),
            $this->generator->generate(new Context($container, false)),
            true
        );
```

File: src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            ->setName('container:generate:js')
+            ->addArgument('access-id', InputArgument::REQUIRED, 'Access id of container that should be mocked.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $accessId = $input->getArgument('access-id');
+        $container = $this->containerRepository->findByAccessId($accessId);
+
+        $this->filesystem->write(
```
Comment (michalsikora) [inline src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php]: this logic should be in separate service
```php

        $container = $this->containerRepository->findByAccessId($accessId);

        $this->filesystem->write(
            sprintf('container-debugger/tests/mock/stg.js', $container->getAccessId()),
            $this->generator->generate(new Context($container, false)),
            true
        );
```

File: src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+     * @var TwigEngine
+     */
+    private $templating;
+
+    /**
+     * @param Filesystem $filesystem
+     * @param TagTreeBuilderInterface $treeBuilder
+     * @param ContainerRepository $containerRepository
+     */
+    public function __construct(Filesystem $filesystem, TagTreeBuilderInterface $treeBuilder, ContainerRepository $containerRepository, TwigEngine $templating)
```
Comment (michalsikora) [inline src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php]: to many dependents here, we can resolve it to move logic from execute method to separate service
information below

File: src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+            ->setName('container:generate:tag-tree')
+            ->addArgument('access-id', InputArgument::REQUIRED, 'Access id of container that should be mocked.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $accessId = $input->getArgument('access-id');
+        $container = $this->containerRepository->findByAccessId($accessId);
+
+        $this->filesystem->write(
```
Comment (michalsikora) [inline src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php]: We can move it to separate service

File: src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php (hunk 5 above)
Comment (kallosz) [inline src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php]: @michalsikora agree. Command shouldn't be aware of this kind of logic.

File: src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php (hunk 6 above)
Comment (kallosz) [inline src/SevenTag/Api/ContainerBundle/Command/JavascriptGeneratorCommand.php]: +1

File: src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php (hunk 7 above)
Comment (kallosz) [inline src/SevenTag/Api/ContainerBundle/Command/TagtreeGeneratorCommand.php]: Same here as @michalsikora mention above.

**Pull request merged**